contracts/utils/GatewayCaller.sol

```diff
@@ -3,15 +3,15 @@
 pragma solidity ^0.8.0;
 
 import { IAxelarGasService } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol';
-import { IAxelarGateway } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGateway.sol';
+import { IAxelarGMPGatewayWithToken } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGMPGatewayWithToken.sol';
 import { IGatewayCaller } from '../interfaces/IGatewayCaller.sol';
 
 /**
  * @title GatewayCaller contract
  * @dev This contract is used to handle cross-chain ITS calls via the Axelar gateway.
  */
 contract GatewayCaller is IGatewayCaller {
-    IAxelarGateway public immutable gateway;
+    IAxelarGMPGatewayWithToken public immutable gateway;
```

Comment:
Same here, cast to legacy functionality when needed

Reply:
done.